x/lockup/keeper/lock.go

```diff
@@ -116,8 +117,18 @@ func (k Keeper) removeTokensFromLock(ctx sdk.Context, lock *types.PeriodLock, co
 
 // AddTokensToLock locks more tokens into a lockup
 // This also saves the lock to the store.
-func (k Keeper) AddTokensToLockByID(ctx sdk.Context, lockID uint64, coins sdk.Coins) (*types.PeriodLock, error) {
+func (k Keeper) AddTokensToLockByID(ctx sdk.Context, lockID uint64, owner sdk.AccAddress, coins sdk.Coins) (*types.PeriodLock, error) {
 	lock, err := k.GetLockByID(ctx, lockID)
+	
+	if lock.GetCoins().Len() != 1 {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, err.Error())
```

Comment:
This error has nothing to do with `err`. You should instead pass a string that alludes to the context of why we're erroring.

Reply:
I am a bit confused as to why this logic check is even required when we already check for coins length during validation check

Reply:
Maybe it isn't? :p

@ValarDragon wdyt?

Reply:
Yeah, sure. If we want tor remove the check, imo we should:

- [ ] Make the function private (rename to `addTokensToLockByID`)
- [ ] Change the function argument from `coins sdk.Coins` to `coin sdk.Coin`

Reply:
how does making function private work? would it just be replacing `AddTokensToLockByID` to `addTokensToLockByID`?

Also if I change the function argument to `coin sdk.Coin` then i'm gonna have to change the rest of the functions that uses expects `coins sdk.Coins` like;` k.bk.SendCoinsFromAccountToModule(ctx, lock.OwnerAddress(), types.ModuleName, coins)` and `k.addTokensToLock(ctx, lock, coins)`, is that expected?

Reply:
and thats how making it private works as well!

Reply:
@ValarDragon tried to make `AddTokensToLockByID(...)` private but got this issue. What else do i need to refactor to accommodate  testing suite? 

<img width="990" alt="Screen Shot 2022-05-03 at 4 22 03 PM" src="https://user-images.githubusercontent.com/26356227/166578221-0a3a225d-a32e-4044-b209-4c2a356962e7.png">